File: 1erdepartamental/python/12.tipo_dato_arreglo/main.py
"""
Programa: Tipo Dato Arreglo
Demuestra colección tipada de objetos personalizados
"""

from core.coleccion import ColeccionProductos, Producto

if __name__ == "__main__":
    inventario = ColeccionProductos()

    inventario.agregar(Producto("Laptop", 15000))
    inventario.agregar(Producto("Mouse", 250))
    inventario.agregar(Producto("Teclado", 800))

    print("=== INVENTARIO ===")
    inventario.mostrar()
    print(f"\nTotal productos: {inventario.cantidad()}")
    print(f"Precio total: ${inventario.precio_total():.2f}")
    print(f"Precio promedio: ${inventario.precio_promedio():.2f}")

File: 1erdepartamental/c++/12.tipo_dato_arreglo/main.cpp
#include <iostream>
#include "core/ColeccionProductos.h"

int main() {
    // Crea coleccion
    ColeccionProductos inventario;
    
    // Agrega productos
    inventario.agregar(Producto("Laptop", 15000));
    inventario.agregar(Producto("Mouse", 250));
    inventario.agregar(Producto("Teclado", 800));
    
    std::cout << "=== INVENTARIO ===" << std::endl;
    inventario.mostrar();
    std::cout << "\nTotal productos: " << inventario.cantidad() << std::endl;
    std::cout << "Precio total: $" << inventario.precioTotal() << std::endl;
    std::cout << "Precio promedio: $" << inventario.precioPromedio() << std::endl;
    
    return 0;
}
